src/ring/middleware/authorization.clj

```diff
@@ -0,0 +1,58 @@
+(ns ring.middleware.authorization
+  (:require [clojure.string :as str]
+            [ring.util.parsing :as parsing]))
+
+(def ^:private re-token68
+  (re-pattern "[A-Za-z0-9._~+/-]+=*"))
+
+(def ^:private re-auth-param
+  (re-pattern (str "(" parsing/re-token ")\\s*=\\s*(?:" parsing/re-value ")")))
+
+(defn- parse-auth-params
+  [auth-params]
+  (reduce (fn [m kv]
+            (if-let [[_ k v1 v2] (re-matches re-auth-param kv)]
+              (assoc m (str/lower-case k) (or v1 v2))
+              m))
+          {}
+          (str/split auth-params #"\s*,\s*")))
+
+(defn- parse-authorization
+  [request]
+  (when-let [[auth-scheme token-or-params]
+             (some-> (get-in request [:headers "authorization"])
+                     (str/split #"\s" 2))]
+    (cond
+      (empty? token-or-params)
+      {:scheme (str/lower-case auth-scheme)}
+
+      (re-matches re-token68 token-or-params)
+      {:scheme  (str/lower-case auth-scheme)
+       :token68 token-or-params}
```

Comment:
While "token68" is used in the RFC as part of the BNF, I think I'd prefer to use `:credentials` as a key. My reasoning is "token68" is the syntax, while "credentials" is the purpose. Let me know if that seems reasonable to you.

Reply:
The RFC describes two alternative syntaxes, "auth-params" and "token68". It uses "credentials" as the umbrella term for everything, including the scheme:
```
     credentials = auth-scheme [ 1*SP ( token68 / #auth-param ) ]
```
([RFC 7235 Page 5](https://datatracker.ietf.org/doc/html/rfc7235#page-5))

Since this middleware will most likely be primarily used by other middleware that implements actual authentication checks based on the data provided, going with the literal names of the RFC seemed to be the right way to me. But I see how that is not very user friendly.

In case the "68" is the part that makes you feel uneasy, dropping it (so the map key becomes "token") would feel OK to me. But I fear that using the RFC's terminology in different ways (using "credentials") could be confusing.

Reply:
The RFC describes two alternative syntaxes, "auth-params" and "token68". It uses "credentials" as the umbrella term for everything, including the scheme:
```
     credentials = auth-scheme [ 1*SP ( token68 / #auth-param ) ]
```
([RFC 7235 Page 5](https://datatracker.ietf.org/doc/html/rfc7235#page-5))

Some authentication schemes pass their "credentials" in the "auth-params" syntax, while others pass them in the "token68" syntax. Since this middleware will most likely be primarily used by developers to create other middleware for actual authentication checks, i.e. people who have to read the RFCs to implement their middleware, going with the literal names of the RFC seemed to be the right way to me. I worry that using the RFC's terminology in different ways (using "credentials" for "token68") could be confusing. Even shortening "token68" to "token" uses a common HTTP RFC name for something else.

If you still prefer "credentials", I can of course change it.

Reply:
No, the RFC's "token68" goes in the `token68` key, the RFC's "auth-params" goes in the `params` key (admittedly I omitted the "auth" prefix used in the RFC, since that seemed redundant). Compare `test-authorization-request` cases "token68" and "auth-params, some malformed".

Reply:
Ah right! I was looking at the code snippet at the top of this commentwithout context and forgot that was the case.

Re-reading it now, I think my logic was that `token68` can contain `base64(username:password )` in the case of basic authentication, which arguably isn't a token, which is why I was thinking `:credentials` as a more general term.

That said, I guess you could argue that it's a token *generated* from the credentials, even if it's a reversible generation. And `:token` is a shorter key than `:credentials`, so perhaps we just go with `:token` then.

Reply:
Changed in https://github.com/ring-clojure/ring-headers/pull/11/commits/9e7d84421fac706ee076947e4fecc98fb1db97b7.